ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLConversationalFlowAgentRunner.java

```diff
@@ -264,8 +265,8 @@ private void processOutput(
         PrivilegedActionException {
         String toolName = getToolName(previousToolSpec);
         String outputKey = toolName + ".output";
-        String outputResponse = parseResponse(output);
-        params.put(outputKey, escapeJson(outputResponse));
+        String outputResponse = filterToolOutput(previousToolSpec.getParameters(), parseResponse(output));
```

Comment:
```
String outputResponse = filterToolOutput(
    previousToolSpec.getParameters() != null ? previousToolSpec.getParameters() : new HashMap<>(), 
    parseResponse(output)
);
```

Reply:
Already fixed in new commit in a different way check ToolUtils.filterToolOutput()